examples/otlp/grpc_metric_main.cc

```diff
@@ -25,24 +28,31 @@ namespace otlp_exporter = opentelemetry::exporter::otlp;
 namespace
 {
 
-otlp_exporter::OtlpGrpcMetricExporterOptions options;
+otlp_exporter::OtlpGrpcMetricExporterOptions exporter_options;
 
 void InitMetrics()
 {
-  auto exporter = otlp_exporter::OtlpGrpcMetricExporterFactory::Create(options);
+  auto exporter = otlp_exporter::OtlpGrpcMetricExporterFactory::Create(exporter_options);
 
   std::string version{"1.2.0"};
   std::string schema{"https://opentelemetry.io/schemas/1.2.0"};
 
   // Initialize and set the global MeterProvider
-  metric_sdk::PeriodicExportingMetricReaderOptions options;
-  options.export_interval_millis = std::chrono::milliseconds(1000);
-  options.export_timeout_millis  = std::chrono::milliseconds(500);
-  std::unique_ptr<metric_sdk::MetricReader> reader{
-      new metric_sdk::PeriodicExportingMetricReader(std::move(exporter), options)};
-  auto provider = std::shared_ptr<metrics_api::MeterProvider>(new metric_sdk::MeterProvider());
-  auto p        = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
-  p->AddMetricReader(std::move(reader));
+  metric_sdk::PeriodicExportingMetricReaderOptions reader_options;
+  reader_options.export_interval_millis = std::chrono::milliseconds(1000);
+  reader_options.export_timeout_millis  = std::chrono::milliseconds(500);
+
+  auto reader =
+      metric_sdk::PeriodicExportingMetricReaderFactory::Create(std::move(exporter), reader_options);
+
+  auto context = metric_sdk::MeterContextFactory::Create();
+  context->AddMetricReader(std::move(reader));
+
+  // FIXME: MeterProvider / MeterProviderFactory should take a unique_ptr
```

Comment:
What is this FIXME about as the implementation is similar to TracerProvider. MeterContext is used across each of the Meters, so passed as shared_ptr.

Reply:
I filed issues #2203 and #2204 for this, the point is that due to spec clarifications, the context should precisely not be shared between providers, for both traces and metrics.

Reply:
Cleaned up the FIXME comments, will be addressed separately by the new issues filed.